### PR TITLE
vma: set refrence count instead of reset and inc

### PR DESCRIPTION
**Pull request by rafiw**
Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/775/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/735/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/784/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/785/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/800/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/801/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/806/ for details (Mellanox internal link).

File: src/vma/dev/rfs_mc.cpp [hunk 1]
```diff
@@ -127,9 +127,8 @@ bool rfs_mc::prepare_flow_spec()
 bool rfs_mc::rx_dispatch_packet(mem_buf_desc_t* p_rx_wc_buf_desc, void* pv_fd_ready_array)
 {
 	// Dispatching: Notify new packet to all registered receivers
-	p_rx_wc_buf_desc->reset_ref_count();
-	p_rx_wc_buf_desc->inc_ref_count();
 
+	p_rx_wc_buf_desc->set_ref_count(1);
```
Comment (OphirMunk) [inline src/vma/dev/rfs_mc.cpp]: Please consider:
```
p_rx_wc_buf_desc->set_ref_count_1();
```

File: src/vma/proto/mem_buf_desc.h [hunk 2]
```diff
@@ -90,6 +90,7 @@ class mem_buf_desc_t {
 	inline void  reset_ref_count() {atomic_set(&n_ref_count, 0);}
 	inline int inc_ref_count() {return atomic_fetch_and_inc(&n_ref_count);}
 	inline int dec_ref_count() {return atomic_fetch_and_dec(&n_ref_count);}
+	inline int set_ref_count(int val) {return atomic_set(&n_ref_count, val);}
```
Comment (OphirMunk) [inline src/vma/proto/mem_buf_desc.h]: Please consider
```
inline int set_ref_count_1() {return atomic_set(&n_ref_count, 1);}
```

Comment (rafiw) [inline src/vma/proto/mem_buf_desc.h]: don't you prefer a more generic function?

**Comment by OphirMunk** [comment]
@rafiw Please open a RM ticket and add its number

**Comment by rafiw** [comment]
958935

**Comment by OphirMunk** [comment]
@rafiw my performance smoke-test shows better results without this PR.
Did you notice the opposite of this?

**Comment by rafiw** [comment]
no i didn't run any pref tests on it

**Comment by OphirMunk** [comment]
@rafiw please show performance improvement for this PR in order to merge it.

File: src/vma/proto/mem_buf_desc.h (hunk 2 above)
Comment (igor-ivanov) [inline src/vma/proto/mem_buf_desc.h]: :+1:  for initial variant as `set_ref_count()`

**Comment by OphirMunk** [comment]
Closed #274 until performance is improved

**Pull request closed without merging**